File: app/src/main/java/org/torproject/android/service/TProxyService.java
package org.torproject.android.service;

public class TProxyService {
    public static native void TProxyStartService(String config_path, int fd);
    public static native void TProxyStopService();
    public static native long[] TProxyGetStats();

    static {
        System.loadLibrary("hev-socks5-tunnel");
    }

    public static final String VIRTUAL_GATEWAY_IPV4 = "198.18.0.1";
    public static final String VIRTUAL_GATEWAY_IPV6 = "fc00::1";
    public final static String FAKE_DNS = "198.18.0.2";

    public static final int TASK_SIZE = 81920;
    public static final int TUNNEL_MTU = 8500;
}

File: app/src/main/java/org/torproject/android/service/TProxyService.kt
package org.torproject.android.service

object TProxyService {
    @JvmStatic
    external fun TProxyStartService(config_path: String?, fd: Int)
    @JvmStatic
    external fun TProxyStopService()
    external fun TProxyGetStats(): LongArray?

    init {
        System.loadLibrary("hev-socks5-tunnel")
    }

    const val VIRTUAL_GATEWAY_IPV4: String = "198.18.0.1"
    const val VIRTUAL_GATEWAY_IPV6: String = "fc00::1"
    const val FAKE_DNS: String = "198.18.0.2"

    const val TASK_SIZE: Int = 81920
    const val TUNNEL_MTU: Int = 8500
}
